### PR TITLE
[api] Add Failover field to Backend

### PR DESCRIPTION
**Pull request by arkodg**
Adds a `failover` field to Backend API so we can support Active/Passive Failove backends within xRoutes similar to https://github.com/envoyproxy/gateway/pull/4033

Relates to https://github.com/envoyproxy/gateway/issues/3055

### CODE REVIEW

**Comment by arkodg** [comment]
cc @alexwo

File: api/v1alpha1/backend_types.go [hunk 1]
```diff
@@ -123,6 +123,15 @@ type BackendSpec struct {
...
 	AppProtocols []AppProtocolType `json:"appProtocols,omitempty"`
+
+	// Failover This indicates whether the backend is designated as a failover.
+	// It is highly recommended to configure active or passive health checks to ensure that failover can be detected
+	// when the active backends become unhealthy and to automatically readjust once the primary backends are healthy again.
+	// The overprovisioning factor is set to 1.4, meaning the failover backends will only start receiving traffic when
+	// the health of the active backends falls below 72%.
+	//
+	// +optional
+	Failover *bool `json:"failover,omitempty"`
```
Comment (guydc) [inline api/v1alpha1/backend_types.go:134]: not imeplemented hide? Or do you plan on implementing this in the near future?

**Review by guydc — APPROVED** [review]

**Review by guydc — APPROVED** [review]

**Comment by ferdinandosimonetti** [comment]
Hello, I'm using the Envoy Gateway version **1.1.2** and (according to `kubectl explain httproute.spec.rules.backendRefs`) the new field is not available.
Any hints on whether the discussed **failover** field will be implemented, and in which official release?
Thanks in advance
Best regards

**Comment by zirain** [comment]
> Hello, I'm using the Envoy Gateway version **1.1.2** and (according to `kubectl explain httproute.spec.rules.backendRefs`) the new field is not available. Any hints on whether the discussed **failover** field will be implemented, and in which official release? Thanks in advance Best regards

should be v1.2

**Comment by ferdinandosimonetti** [comment]
> > Hello, I'm using the Envoy Gateway version **1.1.2** and (according to `kubectl explain httproute.spec.rules.backendRefs`) the new field is not available. Any hints on whether the discussed **failover** field will be implemented, and in which official release? Thanks in advance Best regards
> 
> should be v1.2

If someone wants to test this feature before the **1.2** release, it is possible to do that by installing **v0.0.0-latest** Helm Chart? Thanks again!

**Comment by zirain** [comment]
yes, normally new feature won't be backported.

**Comment by ferdinandosimonetti** [comment]
Sorry. I didn't understand.
I have currently v0.0.0-latest Helm Chart version installed, that took along its CRDs.

But this HTTPRoute 
```
---
apiVersion: gateway.networking.k8s.io/v1
kind: HTTPRoute
metadata:
  name: backend
  namespace: default
spec:
  parentRefs:
    - name: test-fsmn-xyz
      sectionName: https
  hostnames:
    - "test0.fsmn.xyz"
  rules:
    - backendRefs:
        - group: ""
          kind: Service
          name: backend
          port: 3000
        - group: ""
          kind: Service
          name: backend2
          port: 3000
          failover: True
      matches:
        - path:
            type: PathPrefix
            value: /
 ```
 won't apply
 ```
strict decoding error: unknown field "spec.rules[0].backendRefs[1].failover"
```
Did I misunderstand the intended placement of the **Failover:** attribute?

**Comment by arkodg** [comment]
> Sorry. I didn't understand. I have currently v0.0.0-latest Helm Chart version installed, that took along its CRDs.
> 
> But this HTTPRoute
> 
> ```
> ---
> apiVersion: gateway.networking.k8s.io/v1
> kind: HTTPRoute
> metadata:
>   name: backend
>   namespace: default
> spec:
>   parentRefs:
>     - name: test-fsmn-xyz
>       sectionName: https
>   hostnames:
>     - "test0.fsmn.xyz"
>   rules:
>     - backendRefs:
>         - group: ""
>           kind: Service
>           name: backend
>           port: 3000
>         - group: ""
>           kind: Service
>           name: backend2
>           port: 3000
>           failover: True
>       matches:
>         - path:
>             type: PathPrefix
>             value: /
> ```
> 
> won't apply
> 
> ```
> strict decoding error: unknown field "spec.rules[0].backendRefs[1].failover"
> ```
> 
> Did I misunderstand the intended placement of the **Failover:** attribute?

hey the field was renamed to `fallback` and its part of the Backend API, not `backendRef`
https://gateway.envoyproxy.io/latest/api/extension_types/#backendspec
this is an unreleased feature, but you can try it out with `v0.0.0-latest` . 
You'll need to setup health checks in the BackendTrafficPolicy to make sure this works. The docs highlighting this should be out soon

**Pull request merged**